examples/dreamcast/pvr/strided_texture/strided_texture.c

```diff
@@ -0,0 +1,228 @@
+
+/* The height of the texture in pixels */
+#define TEXTURE_HEIGHT   480
+
+/* 
+   Macro to calculate the next power of two for a given integer `x`.
+   
+   Example:
+     NEXT_POWER_OF_TWO(640) -> 1024
+*/
+#define NEXT_POWER_OF_TWO(x)    (1 << (32 - __builtin_clz((x) - 1)))
+
+/* 
+   The power-of-two width to be passed to texture setup functions. 
+   Based on TEXTURE_WIDTH, this should be equal to or greater than 
+   TEXTURE_WIDTH, and will match the hardware requirement for 
+   power-of-two texture dimensions.
+*/
+#define TEXTURE_PADDED_WIDTH    NEXT_POWER_OF_TWO(TEXTURE_WIDTH)
+
+/* 
+   The power-of-two height to be passed to texture setup functions. 
+   Based on TEXTURE_HEIGHT, this should be equal to or greater than 
+   TEXTURE_HEIGHT, following the power-of-two requirement.
+*/
+#define TEXTURE_PADDED_HEIGHT   NEXT_POWER_OF_TWO(TEXTURE_HEIGHT)
+
+/* RGB565 colors for chessboard pattern */
+#define COLOR_BLACK   0x0000
+#define COLOR_WHITE   0xFFFF
+
+static pvr_poly_hdr_t hdr;
+static pvr_vertex_t verts[4];
+
+static pvr_ptr_t board_texture;
+
+static void draw_frame(void) {
+    pvr_wait_ready();
+    pvr_scene_begin();
+
+    pvr_list_begin(PVR_LIST_OP_POLY);
+
+    pvr_prim(&hdr, sizeof(hdr));
+
+    pvr_prim(&verts[0], sizeof(pvr_vertex_t));
+    pvr_prim(&verts[1], sizeof(pvr_vertex_t));
+    pvr_prim(&verts[2], sizeof(pvr_vertex_t));
+    pvr_prim(&verts[3], sizeof(pvr_vertex_t));
+
+    pvr_list_finish();
+    pvr_scene_finish();
+}
+
+static void load_texture(void) {
+    pvr_poly_cxt_t cxt;
+    uint16_t *grid_texture;
+
+    /* Allocate memory for the texture with 32-byte alignment */
+    grid_texture = (uint16_t *)memalign(32, TEXTURE_WIDTH * TEXTURE_HEIGHT * 2);
+
+    /* Generate a chessboard pattern */
+    for(int y = 0; y < TEXTURE_HEIGHT; y++) {
+        for(int x = 0; x < TEXTURE_WIDTH; x++) {
+            /* Determine if we are in an even or odd square. */
+            int square_x = x / 32;
+            int square_y = y / 32;
+            int is_white_square = (square_x + square_y) % 2;
+
+            grid_texture[y * TEXTURE_WIDTH + x] = is_white_square
+                                    ? COLOR_WHITE : COLOR_BLACK;
+        }
+    }
+
+    /* Allocate space in VRAM for the texture */
+    board_texture = pvr_mem_malloc(TEXTURE_WIDTH * TEXTURE_HEIGHT * 2);
+
+    /* Load the chessboard pattern into VRAM */
+    pvr_txr_load(grid_texture, board_texture, 
+                 TEXTURE_WIDTH * TEXTURE_HEIGHT * 2);
+
+    /* Set texture context format for nontwiddled, strided texture */
+    pvr_poly_cxt_txr(&cxt, PVR_LIST_OP_POLY,
+                     PVR_TXRFMT_RGB565 | PVR_TXRFMT_NONTWIDDLED 
+                     | PVR_TXRFMT_X32_STRIDE, TEXTURE_PADDED_WIDTH, 
+                     TEXTURE_PADDED_HEIGHT, board_texture, PVR_FILTER_NONE);
+    pvr_poly_compile(&hdr, &cxt);
+
+    /* Set the STRIDE register to texture width / 32 */
+    PVR_SET(PVR_TXR_STRIDE_MULT, TEXTURE_WIDTH / 32);
```

Comment:
Am I the only one who feels like stuff like this should be dedicated API functionality rather than us just describing how to go in and directly poke registers manually? I feel like this is too low-level to be user-friendly, and the documentation for these registers will be crappier than a dedicated API, plus people will look at the API/docs and not even see this kind of functionality, plus we've talked about deprecating direct access to registers like this...

...plus, this gives us a way to validate the user's input, PLUS it looks like only the first 4 bits are actually used for stride, while the rest of the bits are reserved or used for other things in that register, so you don't really want to just have the user do a raw set anyway.

I would rather see something like:
```
bool pvr_txr_set_stride(unsigned int texture_width);
unsigned int pvr_txr_get_stride(void);
```
Which can then:
    - do the divide by 32 for the user
    - return an error/assert if the dimensions are invalid
    - only sets the first 4 bits of the register
    - lets you retrieve the current value
    - have API documentation for the group that includes a lot of this great commentary in the example which the user would not get just looking at the header file/register map otherwise.

Reply:
Sounds like a great idea! I'll implement it.

Reply:
@gyrovorbis Done